spring-integration-xml/src/main/java/org/springframework/integration/xml/config/XPathMessageSplitterParser.java

```diff
@@ -47,7 +50,11 @@ protected BeanDefinitionBuilder parseHandler(Element element, ParserContext pars
 		Assert.isTrue(hasChild ^ hasReference, "Exactly one of 'xpath-expression' or 'xpath-expression-ref' is required.");
 		if (hasChild) {
 			BeanDefinition beanDefinition = this.xpathParser.parse((Element) xPathExpressionNodes.item(0), parserContext);
-			builder.addConstructorArgValue(beanDefinition);
+			Map<Integer, ConstructorArgumentValues.ValueHolder> indexedArgumentValues =
+					beanDefinition.getConstructorArgumentValues().getIndexedArgumentValues();
+			for (ConstructorArgumentValues.ValueHolder valueHolder : indexedArgumentValues.values()) {
+				builder.addConstructorArgValue(valueHolder.getValue());
+			}
 		}
```

Comment:
This seems a bit dirty - parse and register an `XPathExpressionFactory` but then only use the factory method constructor args.

Perhaps we need to refactor the logic into a utility to add these constructor args to each builder.